usecase/parser/msg.go

```diff
@@ -1294,22 +1295,67 @@ func ParseMsgExec(
 			RawMsgExec: rawMsg,
 		}
 
-		return []command.Command{command_usecase.NewCreateMsgExec(
+		innerCommands := parseMsgExecInnerMsgs(parserParams)
+
+		return append([]command.Command{command_usecase.NewCreateMsgExec(
 			parserParams.MsgCommonParams,
 
 			execParams,
-		)}
+		)}, innerCommands...)
 	}
 
 	execParams := model.MsgExecParams{
 		RawMsgExec: rawMsg,
 	}
 
-	return []command.Command{command_usecase.NewCreateMsgExec(
+	innerCommands := parseMsgExecInnerMsgs(parserParams)
+
+	return append([]command.Command{command_usecase.NewCreateMsgExec(
 		parserParams.MsgCommonParams,
 
 		execParams,
-	)}
+	)}, innerCommands...)
+}
+
+func parseMsgExecInnerMsgs(
+	parserParams utils.CosmosParserParams,
+) []command.Command {
+
+	var commands []command.Command
+	blockHeight := parserParams.MsgCommonParams.BlockHeight
+
+	msgs, ok := parserParams.Msg["msgs"].([]interface{})
+	if !ok {
+		panic(fmt.Errorf("error parsing MsgExec.msgs to []interface{}: %v", parserParams.Msg["msgs"]))
+	}
+
+	for innerMsgIndex, innerMsgInterface := range msgs {
+		innerMsg, ok := innerMsgInterface.(map[string]interface{})
+		if !ok {
+			panic(fmt.Errorf("error parsing MsgExec.msgs[%v] to map[string]interface{}: %v", innerMsgIndex, innerMsgInterface))
+		}
+
+		innerMsgType, ok := innerMsg["@type"].(string)
+		if !ok {
+			panic(fmt.Errorf("error missing '@type' in MsgExec.msgs[%v]: %v", innerMsgIndex, innerMsg))
+		}
+
+		parser := parserParams.ParserManager.GetParser(utils.CosmosParserKey(innerMsgType), utils.ParserBlockHeight(blockHeight))
+
+		msgCommands := parser(utils.CosmosParserParams{
+			AddressPrefix:   parserParams.AddressPrefix,
+			StakingDenom:    parserParams.StakingDenom,
+			TxsResult:       parserParams.TxsResult,
+			MsgCommonParams: parserParams.MsgCommonParams,
+			Msg:             innerMsg,
+			MsgIndex:        parserParams.MsgIndex,
+			ParserManager:   parserParams.ParserManager,
```

Comment:
why we need to pass `ParserManager` here?

Reply:
This is mainly for parsing inner msgs for MsgExec. After parsing MsgExec, we will know its inner msg type, and then will need this `ParserManager` to retrieve the corresponding parser for parsing the inner msg.

Reply:
would the inner messages need `ParserManager`?

Reply:
@davcrypto I am so sorry... I missed the message 🙏  I will always refresh the page later on...

Yes, the inner messages will need the `ParserManager`. As the `ParseMsgExec()` could only parsed out a `model.RawMsgExec`, which does not handle the inner message.

We need to find the inner message type first, and then get the parser function from `ParserManager`.